packages/jazz-tools/src/auth/MagicLinkAuth.ts

```diff
@@ -0,0 +1,256 @@
+
+    (transfer._loadedAs as Account)._raw.core.node.gracefulShutdown();
+
+    const secretSeed = base64URLtoBytes(secret);
+    const accountSecret = this.crypto.agentSecretFromSecretSeed(secretSeed);
+
+    const accountID = cojsonInternals.idforHeader(
+      cojsonInternals.accountHeaderForInitialAgentSecret(
+        accountSecret,
+        this.crypto,
+      ),
+      this.crypto,
+    ) as ID<Account>;
+
+    await this.authenticate({ accountID, accountSecret });
+
+    await this.authSecretStorage.set({
+      accountID,
+      secretSeed,
+      accountSecret,
+      provider: "magicLink",
+    });
+  }
+
+  /**
+   * Accept a transfer from a URL.
+   * @param url - The URL to accept the transfer from.
+   * @param targetHandler - Specifies whether the URL is for consumer or provider.
+   * @returns The accepted MagicLinkAuthTransfer.
+   */
+  public async acceptTransferUrl(
+    url: string,
+    targetHandler: "consumer" | "provider",
+  ) {
+    const { transferId, inviteSecret } = parseMagicLinkAuthUrl(
+      this.options[`${targetHandler}HandlerPath`],
+      url,
+    );
+
+    const account =
+      targetHandler === "consumer"
+        ? await createTemporaryAgent(this.crypto)
+        : Account.getMe();
+
+    const transfer = await account.acceptInvite(
+      transferId,
+      inviteSecret,
+      MagicLinkAuthTransfer,
+    );
+    if (!transfer) throw new Error("Failed to accept invite");
+
+    transfer.acceptedBy = account;
+
+    return transfer;
+  }
+}
+
+/**
+ * Default function to generate a 6-digit confirmation code.
+ * @param crypto - The crypto provider to use for random number generation.
+ * @returns The generated confirmation code.
+ */
+async function defaultConfirmationCodeFn(crypto: CryptoProvider) {
+  let code = "";
+  while (code.length < 6) {
+    // value is 0-15
+    const value = crypto.randomBytes(1)[0]! & 0x0f;
+    // discard values >=10 for uniform distribution 0-9
+    if (value >= 10) continue;
+    code += value.toString();
+  }
+  return code;
+}
+
+const defaultOptions: MagicLinkAuthOptions = {
+  confirmationCodeFn: defaultConfirmationCodeFn,
+  consumerHandlerPath: "/magic-link-handler-consumer",
+  providerHandlerPath: "/magic-link-handler-provider",
+};
+
+/**
+ * Create a temporary agent to keep the transfer secret isolated from persistent accounts.
+ * @param crypto - The crypto provider to use for agent creation.
+ * @returns The created Account.
+ */
+async function createTemporaryAgent(crypto: CryptoProvider) {
+  const [localPeer, magicLinkAuthPeer] = cojsonInternals.connectedPeers(
+    "local",
+    "magicLinkAuth",
+    { peer1role: "server", peer2role: "client" },
+  );
+  Account.getMe()._raw.core.node.syncManager.addPeer(magicLinkAuthPeer);
+
+  const { node } = await LocalNode.withNewlyCreatedAccount({
+    creationProps: { name: "Sandbox account" },
+    peersToLoadFrom: [localPeer],
+    crypto,
+  });
+  return Account.fromNode(node);
```

Comment:
This node should be closed when the flow is completed or fails.

Can't find where this happens